### PR TITLE
fix(DB/Creature): Add missing Blood Guard Porung to Shattered Halls

### PR DESCRIPTION
**Pull request by pklloveyou**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes 

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Ideally we want to fix this encounter properly within the coming weeks.
So there's a couple of problems here, which cannot be handled by SAI alone, such as the targeting method for the Flame Arrows, the spawn of Zealots, the encounter starting from the scout and not from an areatrigger, etc.

One thing we can use is that I thought Porung was a difficulty_entry of the Normal Mode Blood Guard which **is not the case**
Difficulty_entry is DBC and blizz, so we need to use your changes in the creature table

**Comment by pklloveyou** [comment]
Do I need to close it?

**Comment by Gultask** [comment]
Btw, I think it's the file name causing this issue, try renaming using underscores ( _ ) instead of hyphens ( - ) 😁

**Comment by pklloveyou** [comment]
ok

**Comment by Gultask** [comment]
> Do I need to close it?

No need, I'll just ask for some changes. The SAI part can be removed, though

**Comment by pklloveyou** [comment]
You can freely change it.

File: data/sql/updates/pending_db_world/Flame-Arrows.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+DELETE FROM `creature` WHERE `id1` = 17461;
+DELETE FROM `creature` WHERE `id1` = 20923;
+INSERT INTO `creature` VALUES (151283, 20923, 0, 0, 540, 0, 0, 2, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 103320, 0, 0, 0, 0, 0, '', 0);
+INSERT INTO `creature` VALUES (151118, 17461, 0, 0, 540, 0, 0, 1, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 1, 0, 0, 0, 0, 0, '', 46924);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/Flame-Arrows.sql]: 1. Always try to include more than one column when deleting entries, especially in the creature table.
2. Try to add the column names as well. It's unlikely but in the case the table changes structure it's useful to keep the names.
3. Shorten the queries when possible, 1 query runs faster than 10 😛 
```suggestion
DELETE FROM `creature` WHERE `id1` IN (17461, 20923) AND `guid` IN (151118, 151283);
INSERT INTO `creature` (`guid`, `id1`, `id2`, `id3`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES 
(151283, 20923, 0, 0, 540, 0, 0, 2, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 0, 0, 0, 0, 0, 0, '', 46924),
(151118, 17461, 0, 0, 540, 0, 0, 1, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 0, 0, 0, 0, 0, 0, '', 46924);

```

Comment (pklloveyou) [reply data/sql/updates/pending_db_world/Flame-Arrows.sql]: ok

File: data/sql/updates/pending_db_world/Flame_Arrows.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+DELETE FROM `creature` WHERE `id1` = 17461;
+DELETE FROM `creature` WHERE `id1` = 20923;
+INSERT INTO `creature` VALUES (151283, 20923, 0, 0, 540, 0, 0, 2, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 103320, 0, 0, 0, 0, 0, '', 0);
+INSERT INTO `creature` VALUES (151118, 17461, 0, 0, 540, 0, 0, 1, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 1, 0, 0, 0, 0, 0, '', 46924);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/Flame_Arrows.sql:4]:
```suggestion
DELETE FROM `creature` WHERE `id1` IN (17461, 20923) AND `guid` IN (151118, 151283);
INSERT INTO `creature` (`guid`, `id1`, `id2`, `id3`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES 
(151283, 20923, 0, 0, 540, 0, 0, 2, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 0, 0, 0, 0, 0, 0, '', 46924),
(151118, 17461, 0, 0, 540, 0, 0, 1, 1, 1, 512.687, 315.652, 2.0405, 2.98451, 86400, 0, 0, 0, 0, 0, 0, 0, 0, '', 46924);
```

**Comment by Gultask** [comment]
Ty for the PR!

**Review by Gultask — APPROVED** [review]

**Comment by Gultask** [comment]
Right, it's not the title, then. So it's EOF encoding.
I think the fix is to use create_sql.sh to create the sql file

**Comment by Tereneckla** [comment]
It's CR LF instead of LF
Open in for example notepad++ and change the linebreak encoding

**Pull request merged**